packages/core/src/components/field-wrapper/field-wrapper.tsx

```diff
@@ -112,31 +107,53 @@ export class FieldWrapper implements FieldWrapperInterface {
       infoText: this.infoText,
       helperText: this.helperText,
     };
-    const renderStatusSlot = () => {
-      let statusSlot = null;
-      if (this.isValid && this.hasValidContent) {
-        statusSlot = <slot name="valid"></slot>;
-      } else if (this.isInvalid && this.hasInvalidContent) {
-        statusSlot = <slot name="invalid"></slot>;
-      } else if (this.isWarning && this.hasWarningContent) {
-        statusSlot = <slot name="warning"></slot>;
-      } else if (this.isInfo && this.hasInfoContent) {
-        statusSlot = <slot name="info"></slot>;
-      } else if (this.hasHelperContent) {
-        statusSlot = <slot name="helper"></slot>;
-      }
-      return statusSlot;
+    const hasSlotContent = (slotName: string) => {
+      const slot = this.hostElement.querySelector<HTMLSlotElement>(
+        `slot[name="${slotName}"]`
+      );
+      if (!slot) return false;
+
+      const elements = slot.assignedElements();
+      return elements.some((element) => {
+        if (element.textContent?.trim() || element.children.length > 0) {
+          return true;
+        }
+        if (element.tagName.includes('-')) {
+          return true;
+        }
+
+        return false;
```

Comment:
What is the `'-'` check for?
```suggestion
        return (element.textContent?.trim() || element.children.length > 0);
```

Reply:
The '-' check is for when we use empty custom elements. It seems after further testing it was already handled without it, hence I am removed the particular snippet.